static/js/shared/data_fetcher.ts

```diff
@@ -126,257 +84,58 @@ class StatsData {
-        numStatVarsPerPlace[place] = numStatVarsPerPlace[place] + 1;
-      }
-      for (const date in timeSeries.data) {
-        if (date in numOccurencesPerDate) {
-          numOccurencesPerDate[date] = numOccurencesPerDate[date] + 1;
-        } else {
-          numOccurencesPerDate[date] = 1;
-        }
-      }
-    }
-  }
-
-  result.dates = Object.keys(numOccurencesPerDate);
-  result.dates.sort();
-  // Get the number of places that have data, then get the latest date which has data for every stat var for every place with data.
-  // If there's no such date, choose the latest date with data for any stat var
-  const numPlacesWithData: number = places.filter(
-    (place) => numStatVarsPerPlace[place] > 0
-  ).length;
-  let idx = result.dates.length - 1;
-  while (idx >= 0) {
-    const currDate = result.dates[idx];
-    if (
-      numOccurencesPerDate[currDate] ===
-      numPlacesWithData * statsVars.length
-    ) {
-      result.latestCommonDate = currDate;
-      return result;
-    }
-    idx--;
-  }
-  result.latestCommonDate = result.dates[result.dates.length - 1];
-  return result;
-}
-
-/**
- * Returns per capita computation applied to the statVarResponse
+ * Returns per capita computation applied to the time series
  *
  * @param statsVarResponse Response to apply per capita computation to
  * @param populationResponse Population data to use for the per capita computation
  * @returns statVarResponse with the per capita calculation applied to it's values
  */
 function computePerCapita(
-  statsVarResponse: StatApiResponse,
-  populationResponse: StatApiResponse,
+  statSeries: TimeSeries,
+  popSeries: TimeSeries,
   scaling = 1
-): StatApiResponse {
-  const result = { ...statsVarResponse };
-  for (const place in statsVarResponse) {
-    if (!statsVarResponse[place]) continue;
-    const dateValue = statsVarResponse[place].data;
-    const population = populationResponse[place].data;
-    const years = Object.keys(population);
-    years.sort();
-    const yearMin = years[0];
-    const yearMax = years[years.length - 1];
-    for (const date in dateValue) {
-      const year = date.split("-")[0];
-      let pop: number;
-      if (year in population) {
-        pop = population[year];
-      } else if (year < yearMin) {
-        pop = population[yearMin];
-      } else {
-        pop = population[yearMax];
-      }
-      if (pop === ZERO_POPULATION) {
-        result[place].data[date] = ZERO_POPULATION;
-      } else {
-        result[place].data[date] /= pop / scaling;
-      }
+): TimeSeries {
+  const result = _.cloneDeep(statSeries);
+  const years = Object.keys(popSeries.val);
+  years.sort();
+  const yearMin = years[0];
+  const yearMax = years[years.length - 1];
+  for (const date in statSeries.val) {
+    const year = date.split("-")[0];
+    let pop: number;
+    if (year in popSeries.val) {
+      pop = popSeries.val[year];
+    } else if (year < yearMin) {
+      pop = popSeries.val[yearMin];
+    } else {
```

Comment:
Could `year` be between yearMin and yearMax, but not in popSeries.val? In that case, maybe pop should be based off of the latest year in popSeries.val before `year`

Reply:
very good catch! Picking the closest year and added a test.